aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -568,6 +572,35 @@ public synchronized void terminate() throws DataStoreException {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void clear(@NonNull Action onComplete,
+                                   @NonNull Consumer<DataStoreException> onError) {
+        sqliteStorageHelper.close();
+        databaseConnectionHandle.close();
+
+        try {
+            context.deleteDatabase(DATABASE_NAME);
+        } catch (Exception exception) {
```

Comment:
Can we tighted this up at all? `SQLSpecialException | SQLOtherSpecialException sqlError)` ?